### PR TITLE
27/handle db folders

### LINKED ISSUE #27: Handle db folders
On the application launch, create (if it does not exist) the needed folders (in the current case `datasets`).
On "Use Sample Database", remove corresponding files in the VAR_FOLDER.

### PR DESCRIPTION
**Pull request by Painguin**
close #27.

This PR also solves two newly discovered issues:
- The add button would not appear if no dataset was there.
- adding a dataset with the same name as another one would get overwrite it in the filesystem (Thus having two entries that link to the same dataset). The dataset name in the filesystem now corresponds to its id.

### CODE REVIEW

File: public/app/listeners/loadDataset.js [hunk 1]
```diff
@@ -24,13 +23,16 @@ const createNewDataset = ({ id, name, filepath }) => {
 };
 
 const loadDataset = (mainWindow, db) => async (event, args) => {
-  const { fileLocation } = args;
+  const { fileLocation, id } = args;
```
Comment (pyphilia) [inline public/app/listeners/loadDataset.js:26]: I might be wrong, but in `LoadDataset.js` there is no `id` passed as argument. Maybe you can let the id be generated in `createNewDataset` and use the return value to create the `filepath` you need. Be careful because this function is also used in `executePythonAlgorithm.js`

**Review by pyphilia — CHANGES_REQUESTED** [review]
Really nice catching these additional bugs!

File: public/app/listeners/loadDataset.js (hunk 1 above)
Comment (Painguin) [reply public/app/listeners/loadDataset.js:26]: You are right. I missed its use in `executePythonAlgorithm.js` and messed up with the `id` logic... I'll make the changes the way you suggested.

File: public/app/listeners/executePythonAlgorithm.js [hunk 2]
```diff
@@ -33,15 +34,23 @@ const executePythonAlgorithm = (mainWindow, db) => (event, { datasetId }) => {
   process.on('close', (code) => {
     if (code !== 0) {
       logger.error(`python process exited with code ${code}`);
+
+      // delete tmp file
+      if (fs.existsSync(outputPath)) {
```
Comment (pyphilia) [inline public/app/listeners/executePythonAlgorithm.js:39]: What do you think about moving this conditional block after the if-else block? (and remove line 50-51)

**Review by pyphilia — APPROVED** [review]
Nice! I really like that idea of tmp file :P It is handled quite well in the load process.
I'll let @hagoptaminian have a look.

File: src/components/Datasets.js [hunk 3]
```diff
@@ -188,7 +202,12 @@ class Datasets extends Component {
...
 
     if (!datasets.size) {
-      return <Main fullscreen>{t('No datasets available')}</Main>;
+      return (
+        <Main fullscreen>
+          {t('No datasets available')}
+          {this.renderAddButton()}
+        </Main>
+      );
     }
```
Comment (hagoptaminian) [inline src/components/Datasets.js:211]: Why do we use these two conditionals (`!datasets` and `!datasets.size`)? Also, I noticed that while this component is loading, the 'No datasets available' text flashes briefly. This is not a huge deal but is not optimal. We should fix it. I can add it as a separate issue.

**Review by hagoptaminian — CHANGES_REQUESTED** [review]
@Painguin very nice! this works perfectly. I have two main comments:

1. Having worked with these components today I feel we should use the naming `addDataset` rather than `createDataset`, and in the table and database `Added on` instead of `Created at`, etc. Technically we are not creating datasets here and the naming can become confusing for both us and users, especially when we will in fact _create_ datasets with our algorithms later. Please rename the functions/channels/constants in `loadDataset.js`, `Datasets.js`, etc. to become 'added' instead of 'created'.

2. The `Datasets.js` component has become quite big and maybe slightly hard to follow. We should work on breaking it up/refactoring into smaller pieces. I will add this as a separate issue.

**Comment by Painguin** [comment]
As there is a conflict between naming "Added on" for loaded datasets and "Created at" for modified/processed datasets, we will handle the naming issue in issue #23 when both types of datasets will be separated.

**Comment by hagoptaminian** [comment]
@Painguin @pyphilia very nice! Liking how the app is looking. I'll merge this to master.

**Review by hagoptaminian — APPROVED** [review]
